src/app/components/PersistGate.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { PersistGate } from 'redux-persist/es/integration/react';
+
+import { history, persistor, store } from '../store/configureStore';
+import { reloginAction, selectToken } from '../../common/services/user';
+
+const EnhancedPersistGate = ({ children }) => (
+  <PersistGate
+    loading={<div />}
+    onBeforeLift={() => {
+      if (history.location.pathname !== '/auth/logout') {
```

Comment:
I am thinking this logic ahouldnt be here. It should be part of the user service and only call the callback here.

Reply:
You mean the whole handler or just the token check?

Reply:
Yeah, can be moved to the relogin saga.
Also we can connect this component to redux and just dispatch the relogin action.

Reply:
yea lets refactor this, it shouldnt be here